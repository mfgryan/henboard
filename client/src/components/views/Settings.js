// react dep
import React from "react";
import { connect } from "react-redux";
import { Row, Col, Panel } from "react-bootstrap";
import projects from "../../models/projects.js";
import sprints from "../../models/sprints.js";
import user from "../../models/user.js";

// style dep
import "../../css/Settings.css";

export const Planning = ({ project, week, user }) => {
    return (
        <div>
            <Row>
                <Col md={4} mdOffset={2}>
                    <Panel header="project">
                        {project}
                    </Panel>
                </Col>
                <Col md={4}>
                    <Panel header="current Sprint">
                        {week}
                    </Panel>
                </Col>
            </Row>
            <Row>
                <Col md={4} mdOffset={2}>
                    <Panel header="User">
                        {user.name}
                    </Panel>
                </Col>
                <Col md={4}>
                    <Panel header="Email">
                        {user.email}
                    </Panel>
                </Col>
            </Row>
            <Row>
                <Col md={8} mdOffset={2}>
                    <Panel header="Terms">
                        <b>Sprint</b>
                        <p>Timeboxed effort for completing a set of tasks which typically runs for 1-2 weeks.</p>
                        <b>Backlog</b>
                        <p>Used for creating items which have not been assigned to a given sprint.</p>
                    </Panel>
                </Col>
            </Row>
        </div>    
    );
};

const mapStateToProps = (state, ownProps) => {
    let project = projects.getCurrentProject(state);
    let sprint = sprints.getCurrentSprint(state, project.project);
    let id = user.getUser(state);
    return {
        project: project.project,
        week: sprint.week,
        user: id
    };
};

const mapDispatchToProps = (dispatch, ownProps) => {
    return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(Planning);
