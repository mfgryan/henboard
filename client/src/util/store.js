// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate } from "../util/middleware.js";
import data from "../util/data.js";
import rootReducer from "../reducers/index.js";

const store = createStore(rootReducer,applyMiddleware(logger,validate));

export { store }
