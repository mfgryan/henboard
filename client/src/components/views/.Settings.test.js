import React from "react";
import Settings from "./Settings";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import { store } from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it("Settings renders correctly", () => {
    const tree = renderer
        .create(
            <Router>
                <Provider store={store}>
                    <Settings project="henboard" week="01/01/01" />
                </Provider>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
