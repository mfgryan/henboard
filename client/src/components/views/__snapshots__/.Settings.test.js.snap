exports[`test Settings renders correctly 1`] = `
<div
  className="container">
  <div
    className="Header row">
    <div
      className="col-md-4" />
    <div
      className="HeaderDropdownCol col-md-4 col-md-offset-8">
      <div
        className="dropdown btn-group">
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="HeaderDropdownMenu dropdown-toggle btn btn-default"
          disabled={false}
          id="HeaderMenu"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          type="button">
          â‰¡
        </button>
        <ul
          aria-labelledby="HeaderMenu"
          className="dropdown-menu"
          role="menu">
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/home"
              onClick={[Function]}>
              home
            </a>
          </li>
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/backlog"
              onClick={[Function]}>
              backlog
            </a>
          </li>
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/planning"
              onClick={[Function]}>
              planning
            </a>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined} />
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              undo
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}>
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="menuitem"
              tabIndex="-1">
              redo
            </a>
          </li>
          <li
            className="divider"
            onKeyDown={[Function]}
            role="separator"
            style={undefined} />
          <li
            onKeyDown={[Function]}
            onSelect={[Function]}>
            <a
              href="/settings"
              onClick={[Function]}>
              settings
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-4 col-md-offset-4">
      <div
        className="panel panel-default"
        id={undefined}>
        <div
          className="panel-heading">
          project
        </div>
        <div
          className="panel-body">
          henboard
        </div>
      </div>
      <div
        className="panel panel-default"
        id={undefined}>
        <div
          className="panel-heading">
          current Sprint
        </div>
        <div
          className="panel-body">
          05/15/17
        </div>
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-12"
      id="footer">
      <i>
        Henboard 2017
      </i>
    </div>
  </div>
</div>
`;
